packages/back-end/src/routers/saved-queries/saved-queries.controller.ts

```diff
@@ -188,3 +205,227 @@ export async function deleteSavedQuery(
+                }
+              } else {
+                if (table?.numOfColumns) {
+                  const tableSchema = await getInformationSchemaTableById(
+                    context.org.id,
+                    table.id
+                  );
+                  if (!tableSchema) {
+                    // try to fetch the schema if not found:
+                    try {
+                      const tableSchemaData = await fetchOrCreateTableSchema({
+                        context,
+                        datasource,
+                        informationSchema,
+                        tableId: table.id,
+                        databaseName: database.databaseName,
+                        tableSchema: schema.schemaName,
+                        tableName: table.tableName,
+                      });
+                      dbSchemas.set(table.id, tableSchemaData);
+                    } catch (error) {
+                      errors.push(error);
+                    }
+                  } else {
+                    dbSchemas.set(table.id, tableSchema);
+                  }
+                }
+              }
+            }
+          } else {
+            errors.push("no tables found in schema " + schema.schemaName);
+          }
+        }
+      } else {
+        errors.push("no schemas found in database " + database.databaseName);
+      }
+    }
+  } else {
+    errors.push("no databases found in information schema");
+  }
+
+  const schemasString = Array.from(
+    dbSchemas,
+    ([, value]: [string, InformationSchemaTablesInterface]) => {
+      const columnsDescription = value.columns
+        .map((column) => `${column.columnName} (${column.dataType})`)
+        .join(", ");
+      return `Database: ${value.databaseName}, Table: ${value.tableName}, Schema: ${value.tableSchema}, Columns: [${columnsDescription}]`;
+    }
+  ).join("\n");
+
+  const instructions =
+    "You are a data analyst and SQL expert.\n" +
+    "Generate a SQL query to answer the provided question inputted. The resultant query should be ONLY based on the provided context of tables and structure, and return only valid SQL that is executable on the specified data source." +
+    "\n\nTable structure: " +
+    "\n" +
+    schemasString +
+    "\n\nInput: " +
+    input +
+    "\n\nKeep in mind that the table names may be sharded, so you should use the generic table names for tables that have wild cards, such as 'events_*'. If you are querying sharded tables, and you're asked for a date range, you should use something like: \n((_TABLE_SUFFIX BETWEEN '{{date startDateISO \"yyyyMMdd\"}}' AND '{{date endDateISO \"yyyyMMdd\"}}') OR\n" +
+    "   (_TABLE_SUFFIX BETWEEN 'intraday_{{date startDateISO \"yyyyMMdd\"}}' AND 'intraday_{{date endDateISO \"yyyyMMdd\"}}'))\n\n when constructing the query to make sure it's fast and takes use of the sharding. If you do this code, be sure to replace the dates with the correct range.\n";
```

Comment:
Only add these lines about sharding and _TABLE_SUFFIX if the datasource type is `bigquery`.  That is very unique to BigQuery and will just be confusing if used with other engines.

Also, I would add the database type (e.g. `clickhouse` or `postgres`) to the prompt.  Different databases use different syntax, so this is important.